actionpack/lib/action_dispatch/system_test_case.rb

```diff
@@ -120,7 +120,13 @@ def initialize(*) # :nodoc:
       super
       self.class.driver.use
       @proxy_route = if ActionDispatch.test_app
-        Class.new { include ActionDispatch.test_app.routes.url_helpers }.new
+        Class.new do
+          include ActionDispatch.test_app.routes.url_helpers
+
+          def url_options
```

Comment:
let me know if this warrant a test

Reply:
I think it does, given that we’ve already broken it unintentionally once.

Reply:
Done